cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -208,15 +213,19 @@ func (c *CStorPoolController) cStorPoolAddEventHandler(cStorPoolGot *apis.CStorP
 	// IsInitStatus is to check if initial status of cstorpool object is `init`.
 	if IsEmptyStatus(cStorPoolGot) || IsPendingStatus(cStorPoolGot) {
 		// LabelClear is to clear pool label
-		err = pool.LabelClear(cStorPoolGot.Spec.Disks.DiskList)
+		devID, err := c.getDeviceIDs(cStorPoolGot)
+		if err != nil {
+			return string(apis.CStorPoolStatusOffline), errors.Wrapf(err, "failed to get device id of disks for csp %s", cStorPoolGot.Name)
+		}
+		err = pool.LabelClear(devID)
 		if err != nil {
 			glog.Errorf(err.Error(), cStorPoolGot.GetUID())
 		} else {
 			glog.Infof("Label clear successful: %v", string(cStorPoolGot.GetUID()))
 		}
 
 		// CreatePool is to create cstor pool.
-		err = pool.CreatePool(cStorPoolGot)
+		err = pool.CreatePool(cStorPoolGot, devID)
```

Comment:
why do we need to pass devIDs if they are already part of cStorPool?

Reply:
We passed it such as to reduce test code complexity at moment. Have plans to fix it in future PR where I have done more refactoring in pool-mgmt code.